feat: add MVM and MVM-MVI basic compose files

xamcost opened the pull request:
## Description

Hi there! This PR implements docker compose files for:
- the MVM, i.e. just the marketplace without the toolbox. It contains a catalogue instance, but this should be removed before the end of the project, to be replaced by the Sedimark central catalogue. It misses the DLT booth, because the later doesn't provide a Docker image yet (only a Dockerfile so far, but image should come very soon).
- the MVM-MVI, i.e. the marketplace with the Sedimark toolbox: so far, it only includes the NGSI-LD broker from the toolbox. The reason behind this is that I would prefer validating the approach taken in this PR before going further.

The MVM-MVI compose file is built by including the MVM file and the broker compose file: it leverages network isolation by creating (so far) two networks:
- a *sedimark_shared* network, intended to host main components which need to communicate between them.
- a *stellio* network, to host dependencies of the broker.
This approach ensures that no conflict between resources can arise. For instance, the broker uses a postgres database, like many other components in Sedimark. Each of these database will dwell in the dependencies' networks, so we don't have to modify their config to avoid conflicts in the master docker compose.

Anyway, tell me what you think!

## How to test 

A simple `docker compose up -d` in either the `mvm` or `mvm_mvi` folder will pull all the images and start their containers. However, if you want to try the marketplace, you'll need a few things:
- populate the catalogue: for this, you can navigate to `http://localhost:3030` and then create a dataset named `catalogue`, then _add data_, and then upload a file with your catalogue data. You can generate such a file [using this script](https://github.com/Sedimark/catalogue/blob/main/mock_catalogue/create_catalogue.py). Or I can send it to you.
- optional: you can populate the NGSI-LD broker with a dummy asset description. For instance, you can run: 
```bash
curl --request POST \
  --url http://localhost:8080/ngsi-ld/v1/entities \
  --header 'Content-Type: application/json' \
  --header 'Link: <https://sedimark.github.io/broker/jsonld-contexts/sedimark-compound.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
  --data '{
  "id": "urn:ngsi-ld:Asset:01",
  "type": "Asset",
  "title": {
    "type": "Property",
    "value": "Title for Asset 1"
  },
  "creator": {
    "type": "Relationship",
    "object": "urn:ngsi-ld:Participant:MyParticipantId"
  },
  "description": {
    "type": "Property",
    "value": "This is a description for the Asset 1, look at that, im quite fancy"
  },
  "publisher": {
    "type": "Relationship",
    "object": "urn:ngsi-ld:Participant:MyParticipantId"
  },
  "issued": {
    "type": "Property",
    "value": "2024-09-01T11:00:00.000Z"
  },
  "modified": {
    "type": "Property",
    "value": "2024-09-01T11:00:00.000Z"
  },
  "keyword": {
    "type": "Property",
    "value": ["Keyword1", "Keyword3"]
  },
  "license": {
    "type": "Property",
    "value": "https://creativecommons.org/licenses/by-nd/4.0/"
  },
  "spatial": {
    "type": "Property",
    "value": "https://sws.geonames.org/6299418/"
  },
  "startedAtTime": {
    "type": "Property",
    "value": "2024-01-01T01:00:00.000Z"
  },
  "endedAtTime": {
    "type": "Property",
    "value": "2024-12-31T23:59:59.000Z"
  },
  "version": {
    "type": "Property",
    "value": "v1.1"
  },
  "endpointURL": {
    "type": "Property",
    "value": "https://my.endpoint01/url"
  },
  "endpointDescription": {
    "type": "Property",
    "value": "this is what my endpoint do"
  },
  "image": {
    "type": "Property",
    "value": "https://image.com/my/image01"
  },
  "location": {
    "type": "GeoProperty",
    "value": {
      "type": "Point",
      "coordinates": [
        1.5,
        1.5
      ]
    }
  }
}'
```

The marketplace is exposed at `http://localhost:3000`.

## Possible improvements

- use an env file in the Docker compose to set some of the variables. Can be done here on request or in a follow up PR.

JarcauCristian commented on mvm_mvi/docker-compose.yaml:
```diff
@@ -0,0 +1,12 @@
+include:
+  - path: ../mvm/docker-compose.yaml
+  - path:
```
In the mvm_mvi folder it say denied when trying to pull the images with docker compose pull, even though when I am pulling them individually it works.
![image](https://github.com/user-attachments/assets/45e829eb-7214-4a4d-b157-ea02584de4e3)

JarcauCristian replied:
I am logged in into docker and the images for the marketplace and the recommender I cannot pull even though I am logged in.

I think is because the repo is private and I am not the owner of it, but this can be fixed easily.

xamcost replied:
Hi Cristian. The marketplace repo is public but the recommender is indeed private. I added you to WP4 team, which has read access to these repositories, so you should now be able to pull these images. Can you confirm ?

JarcauCristian replied:
Still the same issue, and for the docker compose up in the mvm_mvi, do you know why that does not work, I mean there are images that are public on the docker hub so it should work without the need of a docker login?

JarcauCristian replied:
My bad the issue with running was also from the permission issues with the recommender and the marketplace frontend. Can you add me directly to the repo as a collaborator, if not I think it should work so you can proceed like this for the rest of the integration.

JarcauCristian commented on mvm/docker-compose.yaml:
We can add a env file here as well to be specified in the mvm_mvi folder to make it clear for a user if it wants to modify the environment variables for a specific instance.

xamcost replied:
Implemented in 7b2b1a7ce5a0 . A small note on that: it was easy to add for the MVM only, but it will be a little more tricky for the MVM-MVI, for the reason that many apps have their own config, with overlapping names (for instance, there are two postgres instances for both the AI toolbox and the broker). To avoid conflicts here, I can either override these env variable in the compose file, or we can agree on a naming convention: the latter would be my preferred solution, but I would suggest discussing this in the next meeting we'll have.

JarcauCristian reviewed (changes requested):
The only problem I am seeing is with pulling the images for the mvm_mvi docker compose, but maybe this is a configuration error on my part.

Other than that everything looks ok.

xamcost commented:
Thank you for your quick review @JarcauCristian ! I implemented your suggestions (env file in 7b2b1a7ce5a0 ) and added some basic documentation in the README ( ea1508d9b794 ). Let me know if there are any other changes requested. In any case, there will be follow up PRs, but let's first discuss this with other partners during our upcoming MVM-MVI meeting. I do not intent to push more commits until then, unless requested. What do you think ?

xamcost commented:
Sounds good to me! Thanks for you review @JarcauCristian , merging!
